nsdb-core/src/main/scala/io/radicalbit/nsdb/index/AllFacetIndexes.scala

```diff
@@ -72,15 +72,52 @@ class AllFacetIndexes(basePath: String,
                         rangeFieldName: String,
                         valueFieldName: String,
                         valueFieldType: Option[IndexType[_]],
-                        ranges: Seq[TimeRange]): Seq[Bit] = facetRangeIndex.executeRangeFacet(
-    searcher,
-    query,
-    aggregationType,
-    rangeFieldName,
-    valueFieldName,
-    valueFieldType,
-    ranges
-  )
+                        ranges: Seq[TimeRange]): Seq[Bit] =
+    aggregationType match {
+      case singleAggregation: InternalTemporalSingleAggregation =>
+        facetRangeIndex.executeRangeFacet(
+          searcher,
+          query,
+          singleAggregation,
+          rangeFieldName,
+          valueFieldName,
+          valueFieldType,
+          ranges
+        )
+      case InternalAvgTemporalAggregation =>
+        val sum = facetRangeIndex.executeRangeFacet(
+          searcher,
+          query,
+          InternalSumTemporalAggregation,
+          rangeFieldName,
+          valueFieldName,
+          valueFieldType,
+          ranges
+        )
+
+        val countMap = facetRangeIndex
+          .executeRangeFacet(
+            searcher,
+            query,
+            InternalCountTemporalAggregation,
+            rangeFieldName,
+            valueFieldName,
+            valueFieldType,
+            ranges
+          )
+          .groupBy(_.timestamp)
+
+        sum.map { sumBit =>
+          countMap
+            .get(sumBit.timestamp)
+            .flatMap(_.headOption)
+            .fold(sumBit)(countBit =>
+              sumBit.copy(value = 0.0, tags = Map("count" -> countBit.value, "sum" -> sumBit.value)))
```

Comment:
just to reason: what happen whether for that field I get the SUM but not the COUNT in term of final calculation? The should be the raise of an error or we can provide a defualt value?

Reply:
It's a good point and I can guess it's not clear. By definition, the temporal aggregations are performed against a sequence of time range provided as an input parameter. In the previous statements, the sum and the count gatherings are performed against the same _buckets_. Furthermore, unlike the non-temporal average, it's performed against the same physical index, so it's very unlikely (almost impossible) to find an inconsistency among the sum and the count results